Testing admin_static in admn/base.html

ekivemark opened the pull request:
STATIC_FILES_LOCATION = '/static/'
otherwise in S3 static files get /static appended to the end of the static file name.

aviars commented:
Just FYI we can live without templates/admin right?  ...if removing it fixes the problem.

Best,

Alan

> On Apr 8, 2017, at 5:49 PM, Mark Scrimshire <notifications@github.com> wrote:
> 
> STATIC_FILES_LOCATION = '/static/'
> otherwise in S3 static files get /static appended to the end of the static file name.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/TransparentHealth/hhs_oauth_server/pull/202
> 
> Commit Summary
> 
> Testing admin_static in admn/base.html
> File Changes
> 
> M hhs_oauth_server/settings/aws-dev.py (8)
> Patch Links:
> 
> https://github.com/TransparentHealth/hhs_oauth_server/pull/202.patch
> https://github.com/TransparentHealth/hhs_oauth_server/pull/202.diff
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

ekivemark commented:
I may have fixed it. Still testing but looks promising. It only affected admin pages.

in aws-dev.py which uses S3 I changed

AWS_STORAGE_BUCKET_NAME = 'content-dev-bbonfhir-com'
AWS_S3_CUSTOM_DOMAIN = 's3.amazonaws.com/%s' % AWS_STORAGE_BUCKET_NAME

STATICFILES_LOCATION = 'static'
STATICFILES_STORAGE = 'hhs_oauth_server.s3_storage.StaticStorage'
STATIC_URL = "https://%s/%s/" % (AWS_S3_CUSTOM_DOMAIN, STATICFILES_LOCATION)
# STATIC_URL = '/static/'
print("Static URL:%s" % STATIC_URL)

MEDIAFILES_LOCATION = 'media'
DEAFULT_FILE_STORAGE = 'hhs_oauth_server.s3_storage.MediaStorage'
MEDIA_URL = "https://%s/%s/" % (AWS_S3_CUSTOM_DOMAIN, MEDIAFILES_LOCATION)
# MEDIA_URL = '/media/'

to:

AWS_STORAGE_BUCKET_NAME = 'content-dev-bbonfhir-com'
AWS_S3_CUSTOM_DOMAIN = 's3.amazonaws.com/%s' % AWS_STORAGE_BUCKET_NAME

STATICFILES_LOCATION = '/static/'
STATICFILES_STORAGE = 'hhs_oauth_server.s3_storage.StaticStorage'
STATIC_URL = "https://%s%s" % (AWS_S3_CUSTOM_DOMAIN, STATICFILES_LOCATION)
# STATIC_URL = '/static/'
print("Static URL:%s" % STATIC_URL)

MEDIAFILES_LOCATION = '/media/'
DEAFULT_FILE_STORAGE = 'hhs_oauth_server.s3_storage.MediaStorage'
MEDIA_URL = "https://%s%s" % (AWS_S3_CUSTOM_DOMAIN, MEDIAFILES_LOCATION)
# MEDIA_URL = '/media/'

ie. added '/'  either side of static in STATICFILES_LOCATION

if it is just 'static' then any {% static [staticfilename] %} in an admin file that loads admin_static
wil appear as https://s3.amazonaws.com/content-dev-bbonfhir-com/[staticfile]/static <https://s3.amazonaws.com/content-dev-bbonfhir-com/%5Bstaticfile%5D/static>

- Mark





> On Apr 8, 2017, at 6:14 PM, Alan Viars <notifications@github.com> wrote:
> 
> Just FYI we can live without templates/admin right? ...if removing it fixes the problem.
> 
> Best,
> 
> Alan
> 
> > On Apr 8, 2017, at 5:49 PM, Mark Scrimshire <notifications@github.com> wrote:
> > 
> > STATIC_FILES_LOCATION = '/static/'
> > otherwise in S3 static files get /static appended to the end of the static file name.
> > 
> > You can view, comment on, or merge this pull request online at:
> > 
> > https://github.com/TransparentHealth/hhs_oauth_server/pull/202
> > 
> > Commit Summary
> > 
> > Testing admin_static in admn/base.html
> > File Changes
> > 
> > M hhs_oauth_server/settings/aws-dev.py (8)
> > Patch Links:
> > 
> > https://github.com/TransparentHealth/hhs_oauth_server/pull/202.patch
> > https://github.com/TransparentHealth/hhs_oauth_server/pull/202.diff
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub, or mute the thread.
> > 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/TransparentHealth/hhs_oauth_server/pull/202#issuecomment-292749322>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAGXcMMJW_5IZzqa5pUrV8JHGu5RszVAks5ruAargaJpZM4M313I>.
>
